### PR TITLE
Rename URI prefixes

### LINKED ISSUE #1322: Rename actor prefixes
Follow on from #968: We are going rename the prefixes on our endpoints to more clearly distinguish their intent:
- `/users` => `/app` 
- `/members` => `/gov`
- `/nodes` => `/node`

We will add redirect errors on the old prefixes.

### PR DESCRIPTION
**Pull request by eddyashton**
Fixes #1322.

Renames:
- `/users` => `/app`
- `/members` => `/gov`
- `/nodes` => `/node`

If the old names are used, returns a `308 Permanent Redirect` response pointing to the new prefix. We only do this check on the first node, which I think is safe? Perhaps a mismatch (between "what prefix did they give me" and "what prefix would I like to see") in the pbft or forwarded cases should be an error?

### CODE REVIEW

File: src/http/http_endpoint.h [hunk 1]
```diff
@@ -249,7 +249,10 @@ namespace http
         }
 
         const auto& actor_s = actor_opt.value();
-        auto actor = rpc_map->resolve(actor_s);
+        LOG_DEBUG_FMT("Processing {}", actor_s);
```
Comment (jumaffre) [inline src/http/http_endpoint.h:252]: Is this too verbose?

Comment (eddyashton) [reply src/http/http_endpoint.h:252]: Absolutely! These were just for debugging, removing now.

File: doc/developers/api.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Developer API
 A CCF application is composed of the following:
 
 - The :ref:`Application Entry Point <developers/api:Application Entry Point>` which registers the application in CCF.
-- A collection of :cpp:class:`ccf::EndpointRegistry::Endpoint`s to user HTTP requests and grouped in a single :cpp:class:`ccf::EndpointRegistry`. A :cpp:class:`ccf::EndpointRegistry::Endpoint` reads and writes to the key-value store via the :ref:`Key-Value Store API <developers/kv/api:Key-Value Store API>`.
+- A collection of :cpp:class:`ccf::EndpointRegistry::Endpoint` handling HTTP requests and grouped in a single :cpp:class:`ccf::EndpointRegistry`. A :cpp:class:`ccf::EndpointRegistry::Endpoint` reads and writes to the key-value store via the :ref:`Key-Value Store API <developers/kv/api:Key-Value Store API>`.
```
Comment (achamayou) [inline doc/developers/api.rst:7]:
```suggestion
- A collection of :cpp:class:`ccf::EndpointRegistry::Endpoint`s handling HTTP requests and grouped in a single :cpp:class:`ccf::EndpointRegistry`. A :cpp:class:`ccf::EndpointRegistry::Endpoint` reads and writes to the key-value store via the :ref:`Key-Value Store API <developers/kv/api:Key-Value Store API>`.
```

File: doc/developers/logging_cpp.rst [hunk 3]
```diff
@@ -67,7 +67,7 @@ Each function is installed as the handler for a specific HTTP resource, defined
     :end-before: SNIPPET_END: install_record
     :dedent: 6
 
-This example installs at ``"LOG_record", HTTP_POST``, so will be invoked for requests beginning ``POST /users/LOG_record``.
+This example installs at ``"LOG_record", HTTP_POST``, so will be invoked for requests beginning ``POST /app/LOG_record``.
```
Comment (achamayou) [inline doc/developers/logging_cpp.rst:70]: Let's drop the PREFIX_ stuff now (separate PR).

File: doc/members/open_network.rst [hunk 4]
```diff
@@ -57,7 +57,7 @@ The user can then make user RPCs, for example ``whoAmI`` to retrieve the unique
 
 .. code-block:: bash
 
-    $ curl https://<ccf-node-address>/users/whoAmI --cacert network_cert --key new_user_privk --cert new_user_cert
+    $ curl https://<ccf-node-address>/app/whoAmI --cacert network_cert --key new_user_privk --cert new_user_cert
```
Comment (achamayou) [inline doc/members/open_network.rst:60]: s/whoAmI/user_id?

File: src/http/http_endpoint.h [hunk 5]
```diff
@@ -259,6 +260,27 @@ namespace http
...
+            preferred_actor_s);
+          response.set_header(
+            http::headers::CONTENT_TYPE, http::headervalues::contenttype::TEXT);
+          response.set_body((const uint8_t*)body.data(), body.size());
+
+          auto redirect_url =
+            fmt::format("/{}/{}", preferred_actor_s, rpc_ctx->get_method());
+          response.set_header(http::headers::LOCATION, redirect_url);
+
+          LOG_DEBUG_FMT("Redirecting from deprected '{}' to '{}'", actor_s, redirect_url);
```
Comment (achamayou) [inline src/http/http_endpoint.h:279]:
```suggestion
          LOG_DEBUG_FMT("Redirecting from deprecated '{}' to '{}'", actor_s, redirect_url);
```

**Comment by achamayou** [comment]
@eddyashton I think you missed governance_history.py

File: doc/developers/logging_cpp.rst (hunk 3 above)
Comment (eddyashton) [reply doc/developers/logging_cpp.rst:70]: Yes - I have a branch that mostly does this, just need to make it work for the lua and js apps as well.

File: doc/developers/api.rst (hunk 2 above)
Comment (eddyashton) [reply doc/developers/api.rst:7]: Sure - had to reword slightly to avoid sphinx errors.

**Comment by eddyashton** [comment]
Sphinx is confused by the deprecated methods, printing an error:
```
/data/src/CCF/doc/developers/api.rst:33: WARNING: Error when parsing function declaration.
If the function has no return type:
  Error in declarator or parameters-and-qualifiers
  Invalid C++ declaration: Expected identifier in nested name. [error at 16]
    CCF_DEPRECATED ("HTTP Verb should not be changed after installation: use " "install(...HTTP_POST...)") Endpoint &set_http_post_only()
    ----------------^
If the function has a return type:
  Error in declarator
  If declarator-id with parameters-and-qualifiers:
    Invalid C++ declaration: Expected identifier in nested name. [error at 15]
      CCF_DEPRECATED ("HTTP Verb should not be changed after installation: use " "install(...HTTP_POST...)") Endpoint &set_http_post_only()
      ---------------^
  If parenthesis in noptr-declarator:
    Error in declarator or parameters-and-qualifiers
    If pointer to member declarator:
      Invalid C++ declaration: Expected identifier in nested name. [error at 16]
        CCF_DEPRECATED ("HTTP Verb should not be changed after installation: use " "install(...HTTP_POST...)") Endpoint &set_http_post_only()
        ----------------^
    If declarator-id:
      Invalid C++ declaration: Expected identifier in nested name. [error at 16]
        CCF_DEPRECATED ("HTTP Verb should not be changed after installation: use " "install(...HTTP_POST...)") Endpoint &set_http_post_only()
        ----------------^
```

and pasting the entire block as the function signature:
image

I've just removed the doxyen from these deprecated functions so they don't show up in the docs at all. Next time around we may have to work out how to doxygenate the deprecation.

**Pull request merged**